pkg/distributor/distributor.go

```diff
@@ -1028,6 +1037,59 @@ func (d *Distributor) prePushForwardingMiddleware(next push.Func) push.Func {
 	}
 }
 
+// prePushEphemeralMiddleware is used as push.Func middleware in front of push method.
+// If marking series as ephemeral is enabled, this middleware uses the ephemeral series
+// provider to determine whether a time series should be marked as ephemeral.
+func (d *Distributor) prePushEphemeralMiddleware(next push.Func) push.Func {
+	if !d.cfg.EphemeralSeriesEnabled {
+		return next
+	}
+
+	return func(ctx context.Context, pushReq *push.Request) (*mimirpb.WriteResponse, error) {
+		userID, err := tenant.TenantID(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		req, err := pushReq.WriteRequest()
+		if err != nil {
+			return nil, err
+		}
+
+		req.EphemeralTimeseries = nil
```

Comment:
Why is this required? Add a comment to explain it, please.

Reply:
added a comment: 9d861299f